### ISSUE TITLE
Overly worrysome "useless echo?"

### ISSUE THREAD

**Issue by arifd** [koalaman/shellcheck#2336]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): 
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:
```bash
#!/bin/bash

# adds two numbers ($1 and $2)
add() {
  echo "$(echo "${1}+${2}" | bc)"
}

echo "$(add 1 1)"
```

#### Here's what shellcheck currently says:
```
Line 5:
  echo "$(echo "${1}+${2}" | bc)"
       ^-- SC2005: Useless echo? Instead of 'echo $(cmd)', just use 'cmd'.
 
Line 8:
echo "$(add 1 1)"
     ^-- SC2005: Useless echo? Instead of 'echo $(cmd)', just use 'cmd'.
```

#### Here's what I wanted or expected to see:

Both echos to me seem necessary. and perhaps obviously so? I need the echo in line 5 to use the pipe and in line 8 because I desire to have that command substitution... well, echoed!

Should shellcheck be made to be aware of these legitimate uses?

**Comment by koalaman** [comment]
ShellCheck is right, these are both unnecessary and the code could/should be written like this:

```sh
#!/bin/bash
# adds two numbers ($1 and $2)
add() {
  echo "${1}+${2}" | bc
}
add 1 1
```

It doesn't make a significant difference in this particular case since `bc`'s output is small and it doesn't have a useful exit status on bad input, but it's just easier and works the same. If you refresh the [the wiki](https://github.com/koalaman/shellcheck/wiki/SC2005), I included more cases where avoiding the `echo` *does* make a significant difference to speed and correctness.

**Comment by arifd** [comment]
Wow, crap! I feel a little stupid... I assumed the echo gets used up in the pipe and the substitution returns a value, but you haven't told the function to return any value. And the same outside the function.

So ... Thank you! and Thanks Shellcheck! :D

**Comment by koalaman** [comment]
No problem, discovering these things is what ShellCheck is for ^^
